### PR TITLE
Add service principle check

### PR DESCRIPTION
**Pull request by mrzta**
This is to check for service principle expiry
and send a notification on slack

### CODE REVIEW

File: .github/workflows/check_sp.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        if: fromJson(steps.pwsh_check_expire.outputs.json_data).data.Alert
+        uses: rtCamp/action-slack-notify@master
+        env:
+          SLACK_COLOR: ${{env.SLACK_ERROR}}
+          SLACK_TITLE: A Service Principal secret is expiring soon
+          SLACK_MESSAGE: |
+            The Service Principal *${{ fromJson(steps.pwsh_check_expire.outputs.json_data).data.Application }}*
+            secret *${{ fromJson(steps.pwsh_check_expire.outputs.json_data).data.Name }}* is due to expire in *${{fromJson(steps.pwsh_check_expire.outputs.json_data).data.ExpiresDays}}* days.
+            Please follow the <https://dfe-technical-guidance.london.cloudapps.digital/infrastructure/hosting/azure-cip/#use-the-service-principal-in-external-systems|Use the service principal in external systems> process to renew.
+            This secret is used by all BAT apps so be sure to update the publish, find and register secrets too.
```
Comment (saliceti) [inline .github/workflows/check_sp.yml:51]: This is unrelated

File: documentation/slack-webhook-integration.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+### Step 1: Create a Webhook in Slack
```
Comment (saliceti) [inline documentation/slack-webhook-integration.md:2]: I don't think we should use this technique as it's not supported by Slack anymore.
This is the official way: https://github.com/DFE-Digital/technical-guidance/pull/183

File: .github/workflows/check_sp.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          ExpiresWithinDays: 30
+
+      - name: Slack Notification
+        if: fromJson(steps.pwsh_check_expire.outputs.json_data).data.Alert
+        uses: rtCamp/action-slack-notify@master
+        env:
+          SLACK_COLOR: ${{env.SLACK_ERROR}}
+          SLACK_TITLE: A Service Principal secret is expiring soon
+          SLACK_MESSAGE: |
+            The Service Principal *${{ fromJson(steps.pwsh_check_expire.outputs.json_data).data.Application }}*
```
Comment (smneal) [inline .github/workflows/check_sp.yml:48]: I think it would be good to have a link to the documentation that explains how to resolve the alert.

Comment (mrzta) [reply .github/workflows/check_sp.yml:48]: added another section 3

**Comment by mrzta** [comment]
[P - Add SP check for s189 registrations](https://trello.com/c/HuvJN6eg/531-p-add-sp-check-for-s189-registrations)

**Review by smneal — APPROVED** [review]
